src/lighteval/models/transformers/transformers_model.py

```diff
@@ -221,11 +235,21 @@ def get_model_sha(self):
         return _get_model_sha(repo_id=self.pretrained, revision=self.revision)
 
 
-class BaseModel(LightevalModel):
+@dataclass
+class BaseModelConfig(TransformersModelConfig):
+    def __post_init__(self):
+        super()
+
+        logger.warning(
+            "Careful, BaseModelConfig is deprecated and will be removed, you should use TransformersModelConfig instead!"
+        )
```

Comment:
Better raise a deprecation warning:
```suggestion
        warnings.warn(
            "BaseModelConfig is deprecated and will be removed. Use TransformersModelConfig instead",
            FutureWarning,
        )
```
You need to `import warnings` at the top of the module.

Reply:
Ig we go with this approach it would be nice to have: `logging.captureWarnings(capture=True)` so that the warnings are handle by the logging system

Reply:
The difference is: https://docs.python.org/3/howto/logging.html#when-to-use-logging
> - [warnings.warn()](https://docs.python.org/3/library/warnings.html#warnings.warn) in library code if the issue is avoidable and the client application should be modified to eliminate the warning
> - A logger’s [Logger.warning()](https://docs.python.org/3/library/logging.html#logging.Logger.warning) method if there is nothing the client application can do about the situation, but the event should still be noted

Reply:
In that case setting `logging.captureWarnings(capture=True)` would be better !

Reply:
Where would you want us to set the capture @NathanHB ?

Reply:
for better homogeinity when logging out messages

Reply:
I think `__main__.py` would be the best place